### ISSUE TITLE
Upon helm execution "an error occurred forwarding 39967 -> 44134"

### ISSUE THREAD

**Issue by nerdguru** [apache/incubator-openwhisk-deploy-kube#229]
Hi there,

7 days in, I may be the first person not on the development team to try this out and I'm a helm noob, so keep that in mind.  I got through the Helm install and the "Initial setup" cleanly for an attempt at putting OpenWhisk on top of a minikube cluster.  For the "Customize the Deployment" section, I first verified that nothing was already listening to 31001 (with netstat -no | grep 31001, which returned nothing) and I substituted my local IP address for the example in mycluster.yaml.

First minor thing, the instructions don't tell you to cd into the helm directory, so the first time I tried to run "helm install . --namespace=openwhisk --name=owdev -f mycluster.yaml" in my home directory, helm complained that it couldn't find a Chart.yaml.  As a helm noob, I assumed it would traverse my folders until it found one, but I guess not.

After cd'ing to the helm directory that contains the Chart.yaml and copying the aforementioned mycluster.yaml into that directory, here's what I get:

```
$ helm install . --namespace=openwhisk --name=owdev -f mycluster.yaml
E0620 10:37:57.405197   26380 portforward.go:331] an error occurred forwarding 39967 -> 44134: error forwarding port 44134 to pod 1fe0b89e31e42c806fef7c73a5393f4fcf41d76262e64500e7f1d73c44abc871, uid : unable to do port forwarding: socat not found.
Error: transport is closing
```

I have no idea where to start troubleshooting that since none of the ports mentioned are in either the Chart.yaml or the mycluster.yaml.  I had a prior OpenWhisk install attempt using the old instructions on this same minikube cluster, but removed all the pods/services/deployments.  Not sure if that's impacting what I'm seeing here.

Any clues/thoughts?

I love the approach you're taking here and look forward to getting it running!

**Comment by dgrove-oss** [comment]
Thanks for trying it out and the suggested improvements to the docs.  I will add the `cd` command (missed that when I moved the instructions from helm/README.md to the top-level directory).

Check to see if `tiller-deploy` is running in your minikube cluster via `kubectl get pods -n kube-system`   When I restructured the READMEs, I may have made it to easy to overlook the commands you need to use configure helm/tiller by moving them into `docs/helm.md`

**Comment by nerdguru** [comment]
Hi David,

Thanks so much for the quick reply!  Certainly understandable on the `cd`. Carlos warned me this was early stages but like I said, I love the direction here and thought I'd try it out.

As instructed, here's my output:

```
$ kubectl get pods -n kube-system
NAME                                    READY     STATUS    RESTARTS   AGE
etcd-minikube                           1/1       Running   1          16d
kube-addon-manager-minikube             1/1       Running   1          16d
kube-apiserver-minikube                 1/1       Running   1          16d
kube-controller-manager-minikube        1/1       Running   1          16d
kube-dns-86f4d74b45-58wqj               3/3       Running   2075       16d
kube-proxy-lsrx4                        1/1       Running   1          16d
kube-scheduler-minikube                 1/1       Running   1          16d
kubernetes-dashboard-5498ccf677-mqvn9   1/1       Running   2          16d
storage-provisioner                     1/1       Running   2          16d
tiller-deploy-f9b8476d-zkwfk            1/1       Running   0          6h
```

The helm docs you provided were easy to find, actually.  They're linked off the main Readme.md and I performed the permissions command on Step 5 there prior to trying the `helm install` that gave me troubles.

If you'd like to get on a Webex or something on Thursday to make this easier to diagnose in real time, feel free to contact me at petercjo@cisco.com with some times that work.  My guess would be there's some versioning incompatibility or a missed step here somewhere.

**Comment by csantanapr** [comment]
Hi @nerdguru can you provide some versions
What OS are you using?
kubectl 
```
kubectl version
```
minikube
```
minikube version
```

**Comment by csantanapr** [comment]
@nerdguru here is a blog that might have some help, check it out
https://medium.com/openwhisk/deploying-openwhisk-on-kubernetes-3f55f781fbab

**Comment by nerdguru** [comment]
@csantanapr Thanks for that link, very nicely put together.  Here's mine:

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

$ minikube version
minikube version: v0.27.0
```

TL;DR
I'm on later versions of both K8S and minikube than Priti's excellent writeup.

Would you recommend I start over with minikube on 1.9?  Or is there something else I should look at to try to salvage 1.10?

**Comment by dgrove-oss** [comment]
I was attempting (on MacOS) to get minikube 0.28 and kubernetes 1.10.4 to work for OpenWhisk the end of last week.  So far, no joy (ie it doesn't work for me yet, but I haven't totally given up on it yet either).

I believe I have gotten the combination of minikube 0.25.2 and kubernetes 1.10.0 to work together (also on MacOS), but I will try to verify that today and report back.

**Comment by dgrove-oss** [comment]
Circling back.  I'm having problems with kube-dns not starting properly on the combination of 0.25.2/1.10.0.   

@nerdguru, looking at your pod listing, I'm a little wary of 
`kube-dns-86f4d74b45-58wqj               3/3       Running   2075       16d`
that looks like maybe the kube-dns service is having problems.  Over 2000 restarts seems really fishy.

Maybe try a known good version combination like minikube 0.25.2/ kubernetes 1.9.0 just to make sure you can get that to work, then explore more modern versions on linux.

**Comment by nerdguru** [comment]
Sounds good @dgrove-oss.  I'll try that combination and let you know how it turns out.

**Comment by nerdguru** [comment]
Sorry it took me a couple of days to get back to this, but I wiped my NUK (and bare metal may be my problem as we'll soon see) with Ubuntu 16.04 and started from scratch, but got similar results.

Version info:

`# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.4 LTS
Release:	16.04
Codename:	xenial

# docker version
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64
 Experimental: false

# kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.4", GitCommit:"bee2d1505c4fe820744d26d41ecd3fdd4a3d6546", GitTreeState:"clean", BuildDate:"2018-03-21T21:48:36Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}

# minikube version
minikube version: v0.25.2
`

As advised, I started following this: https://medium.com/openwhisk/deploying-openwhisk-on-kubernetes-3f55f781fbab

but made Ubuntu substitutions as needed for things like installing asdf.  I launched minikube with:

```
# minikube start --vm-driver=none
There is a newer version of minikube available (v0.28.0).  Download it here:
https://github.com/kubernetes/minikube/releases/tag/v0.28.0

To disable this notification, run the following:
minikube config set WantUpdateNotification false
Starting local Kubernetes v1.9.4 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Downloading localkube binary
 163.02 MB / 163.02 MB [============================================] 100.00% 0s
 0 B / 65 B [----------------------------------------------------------]   0.00%
 65 B / 65 B [======================================================] 100.00% 0sSetting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
Kubectl is now configured to use the cluster.
```

When I get to the OpenWhisk install, I see a similar error to before:
```
# helm install . --namespace=openwhisk -f mycluster.yaml
E0627 12:56:26.552804   18126 portforward.go:331] an error occurred forwarding 42989 -> 44134: error forwarding port 44134 to pod 08edc78aca326c3168069b3cc9f4235a5c2dce034cb9286a7a1b67f80e1c1d25, uid : unable to do port forwarding: socat not found.
```

Here's the running pods:

```
# kubectl get pods -n kube-systemNAME                                    READY     STATUS    RESTARTS   AGE
kube-addon-manager-k8s                  1/1       Running   0          3h
kube-dns-54cccfbdf8-n886f               3/3       Running   0          3h
kubernetes-dashboard-77d8b98585-tj8hv   1/1       Running   0          3h
storage-provisioner                     1/1       Running   0          3h
tiller-deploy-7ccf99cd64-wkhjg          1/1       Running   0          3h
```

So, cleaner install than last time and DNS is healthy, unlike before.

Thoughts?  The no-vm driver maybe and I need to try this on a VM instead of bare metal?

**Comment by dgrove-oss** [comment]
hi, I was able to successfully run OpenWhisk on minikube on something similar to your environment this afternoon.  Maybe there is some useful version information here.

I have a bare metal x86_64 server running ubuntu 16.04.4 LTS.  I installed asdf (just because I am used to it), and then used asdf to install minikube 0.25.2 and kubectl 1.9.8.   For a driver, I installed virtualbox (via apt-get).   With this combination, I was able to go through the steps of configuring minikube, installing helm (picked up latest version 2.9.1), and bringing up openwhisk on my minikube cluster running in virtualbox.

**Comment by nerdguru** [comment]
Thanks Dave for the intel.  I tried it on my MacBook Air with Minikube and was able to get all the pods to come up, although I couldn't get them to respond given the lack of CPU and Memory I was able to give the VM, so my problem not yours.  I'm going to try this on GKE tomorrow but I'll close this because I think we've now proven this is a no-vm minikube issue, not an OpenWhisk issue.

Thanks so much for the responsiveness!
